### ISSUE TITLE
Incompatibility with Intake-ESM v2022.9.18

### ISSUE THREAD

**Issue by rmshkv** [klindsay28/analysis_tools#9]
`cases_metadata_to_catalog()` in `catalog_utils.py` is incompatible with the most recent version of Intake-ESM, v2022.9.18. When the function tries to call [`intake.open_esm_datastore()`](https://github.com/klindsay28/analysis_tools/blob/08bbee9dbc66e0f7486a01d45497d4bbca0ad06f/analysis_tools/catalog_utils.py#L108), the following error is thrown: `TypeError: esm_datastore.__init__() takes 2 positional arguments but 3 were given` despite 2 arguments being given. The last version of Intake-ESM confirmed to work without this issue is v2021.8.17.

**Comment by andersy005** [comment]
@rmshkv, 

https://github.com/klindsay28/analysis_tools/blob/08bbee9dbc66e0f7486a01d45497d4bbca0ad06f/analysis_tools/catalog_utils.py#L108

should be changed to 

```python
return intake.open_esm_datastore({'df': esmcol_data, 'esmcat': esmcol_spec})
```

**Comment by andersy005** [comment]
Depending on whether you want to maintain backward compatibility, you may support versions older than `v2022.9.18` as follows:
 

```python

import packaging.version
import intake_esm

.... 
....
....


if packaging.version.Version('2022.9.18') < packaging.version.Version(intake_esm.__version__):
    return intake.open_esm_datastore(esmcol_data, esmcol_spec) 
else:
    return intake.open_esm_datastore({'df': esmcol_data, 'esmcat': esmcol_spec})

```

**Comment by andersy005** [comment]
@rmshkv, i found a bug in my suggestions, and just updated the above comments to reflect the correct versions

**Comment by klindsay28** [comment]
fixed by #10
